x-pack/plugins/endpoint/common/types.ts

```diff
@@ -24,10 +29,7 @@ export type ImmutableMap<K, V> = ReadonlyMap<Immutable<K>, Immutable<V>>;
 export type ImmutableSet<T> = ReadonlySet<Immutable<T>>;
 export type ImmutableObject<T> = { readonly [K in keyof T]: Immutable<T[K]> };
 
-export enum Direction {
-  asc = 'asc',
-  desc = 'desc',
-}
+export type Direction = 'asc' | 'desc';
```

Comment:
@oatkiller Is `type` better than `enum`? TypeScript n00b here.

Reply:
Other plugins have used enum for this... for example, siem. So just curious what the advantages/disadvantages are.

Reply:
I didn't have a strong reason for changing this, was just getting around some quick type errors. Let me try and put it back.

That being said, here's what I know about the diff between a union type and an enum:
https://www.typescriptlang.org/play/#code/PTAEGECcFMEMBdoGdS1JArgO3gSwLbSgBusANhtAHQBQ0WG+oAIrjAMZ4D2WoA3jVBDUSdqAC8oAIwAaUCFABPLhlDtYvdgAsNAcyLwtRTDgJFSFIgHdgXNTqy7cj0IYOKADtEHCAJsjFJAHJ1eCQgmgBfGhoFAFFIWCRoX1R4dGw8QloFXAAzJRVQKw1EVPg7JC8yMlAgpPYgkTqG+l9nXSayBGhIGViweE9kYpUyVLzYXFqNVOUMINSsaBTXOxgPbvYiclrnJHgNbaQB0Gc1Ln8aIa8WNmhOXB4AQTIkO2CGpoAfOv9RCKxABUQMEQNAAElQEDDM4ANbg+iMUYYcagABGRF0XEuoDyXEgoAOGDyBTIuDhRHRzlgkEUoCMsH8kBO0OKWlw2lQMFAWC46Ss0BqoDhfKsWFobIAmkV1Lx8LAPKAPDBkukLJQUBUhFhYIRysMULNQBhkq4jPhJeCIeloKSHnhiEL6Zj2FxCEblRtIFxju9IABaA6KMhEN1YA6lK3AGLAEGgMGgADK9DwyzeJuSqXObgZcGZoC4BTQTgKeWm0AAXGCY0imABxfIp0zplACIQKZ68AiwfSgAAUzngPt8GG2qXR9IADAAPABM4DkGlAzyT4AhULd+AVdRkQQAlD4e33gnv+j5O7xoDPEBGnrx0WRfXCB0OR2PVpPQLO57IL2Bl1XdcoWvdhugVbheA8LghzqABCA8fGvW8kHvAAhJ92BfYIEPPDswAAFSMVxEgrQl+zQVDHFDDFFEQQsCg1IhZwAZjQpdeCAjciWgfBOS4J9eCCABuA9+nw9lOS0IktDGCcDGI7oDlo+ii3NIhy1DSUSKmUNCWCUS8PkMBjyIf52EgXAPAqSAjwVfRmACSzrIJCQ6h-cAgiM3J7KIKQpynUB8AJIg1LcZIoiAA

Reply:
so i tried to go back and make everything the enum, but it turns out we have a whole lot of places in the app writing 'asc' and 'desc' as literals. I don't honestly know what the best practice is, but for the sake of keeping the (runtime) enum out of the validation that will be used on FE and BE, I'm going to keep this as a type for now. I'm aware that this may not be right, I just don't have quite enough experience w/ this yet.

Reply:
Nah, that's fine. I imagine that could be difficult to work into the schema validation. Let's come back to it later.